Update colors and typography of default Learning Mode template

donnapep opened the pull request:
Resolves #6736 (just colors & typography, spacing & alignment will be added to a new issue).

## Proposed Changes
* Updates the colors and typography for class themes for the default Learning Mode template.
* Adds a new `colors.scss` file where we can collect colors that are used in multiple places in LM. I thought about making these CSS variables but I'm not sure if that's better?

## Notes
* The CSS in `mobile.scss` is not mobile first, but that's because they were already using `max-width` instead of `min-width`, and I didn't want to break anything by changing it.
* ~Course title isn't styled because I need the changes in https://github.com/Automattic/sensei/pull/6762.~ Course title is now styled.

## Testing Instructions
<!--
Add as many details as possible to help others reproduce the issue and test the changes.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Enable Divi or Astra.
2. Enable the default Learning Mode template.
3. View a lesson in Learning Mode.
4. Check that the colors and typography match the designs.
5. Check out [this branch](https://github.com/Automattic/themes/pull/7008).
6. Enable the Course theme.
7. View a lesson in Learning Mode.
8. Check that the colors and typography match the designs.

## Pre-Merge Checklist
<!-- Complete applicable items on this checklist **before** merging. Items that are not applicable can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed. -->
- [x] PR title and description contain sufficient detail and accurately describe the changes
- [x] Acceptance criteria is met
- [x] Decisions are publicly documented
- [x] Adheres to coding standards ([PHP](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/), [JavaScript](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/), [CSS](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/), [HTML](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/html/))
- [x] All strings are translatable (without concatenation, handles plurals)
- [x] Follows our naming conventions (P6rkRX-4oA-p2)
- [x] Hooks (p6rkRX-1uS-p2) and functions are documented
- [ ] New UIs are responsive and use a [mobile-first approach](https://zellwk.com/blog/how-to-write-mobile-first-css/)
- [x] New UIs match the designs
- [x] Different user privileges (admin, teacher, subscriber) are tested as appropriate
- [x] Code is tested on the minimum supported PHP and WordPress versions
- [x] User interface changes have been tested on the latest versions of Chrome, Firefox and Safari
- [x] "Needs Documentation" label is added if this change requires updates to documentation
- [x] Known issues are created as new GitHub issues

donnapep committed:
Update class theme styles for default Learning Mode template

donnapep committed:
Fix linter

donnapep committed:
Merge branch 'trunk' into update/learning-mode-default-template

merkushin commented:
On desktop looks pretty good.

Lesson title (h1)
- line-height: expected 64px, actual 72.8px.

Complete Lesson button
- line-height: expected 48px, actual 18px.

Take Quiz button
- line-height: expected 48px, actual 18px.

Current lesson in the Course Navigation:
- line-height: expected auto, actual 22.4px (set as 1.6 in CSS).

Quiz link in the Course Navigation:
- line-height: expected auto, actual 20.8px (set as 1.6 in CSS).

Lesson with prerequisite in the Course Navigation:
- line-height: expected auto, actual 22.4px (set as 1.6)

Same issue (expected auto, actual is set to 1.6) with the course title, course progress and exit course link.

Not sure if it is an issue.

As for colors, Complete Lesson and Take Quiz buttons have different hover colors, which might be okay. But the hover color for the Complete Lesson button seemed a bit weird for me, like it came from another palette. At the same time, I didn't find any guidelines for them.

Maybe @Luchadores would suggest on hover colors for buttons?

Luchadores commented:
Hi @merkushin 👋 

I have updated the Figma file [here](https://www.figma.com/file/8mY8ui4zYK1Ylw2yIFi9Lx/Learning-Mode?node-id=20020045%3A37699&t=MelhwA7rLALnrmDv-1) with all the button styles we have. 

**Default**
<img width="1400" alt="image" src="https://user-images.githubusercontent.com/10038976/231424784-0072af40-a0a9-424d-b314-4c1e4d573123.png">

**Course Theme Variation 1**
<img width="1858" alt="image" src="https://user-images.githubusercontent.com/10038976/231424918-9ed759dd-388a-4f3e-b1a6-3c30b88c93c6.png">

**Course Theme Variation 2**
<img width="1857" alt="image" src="https://user-images.githubusercontent.com/10038976/231424998-1753799a-ddde-4dd6-9b52-f00f7f57a40a.png">

**Course Theme Variation 3**
<img width="1854" alt="image" src="https://user-images.githubusercontent.com/10038976/231425084-b9b63aa2-14fe-4826-83e3-beaf3b5441a9.png">

**Course Theme Variation 4**
<img width="1842" alt="image" src="https://user-images.githubusercontent.com/10038976/231425171-125e115c-2f9d-49c0-933f-4b540dcd895d.png">

There is also a screen that explains the mapping of all the CTAs. @donnapep might know about this one, as we used it back when we developed the Course theme. 
<img width="1009" alt="image" src="https://user-images.githubusercontent.com/10038976/231425667-d0714b85-aed3-471e-8b8e-c7e392fbff39.png">


Hope this helps. Let me know if there are any other questions that are coming up.

donnapep commented:
@merkushin Since line-height relates to spacing, I didn't follow the designs to spec, except for the buttons and a few small things. Also, line-height may not be the appropriate CSS property to use. It could be padding or margin as I did for the buttons. And when using line-height, we should use unitless values, not pixels, so that it's more responsive. Figma doesn't always output the best CSS, so we shouldn't rely on it 100%.

Regardless, I'll work spacing out in the next PR. Will look into the hover color designs to see if they match on classic themes. The Course theme hover colors should be to spec already.

donnapep committed:
Merge branch 'feature/learning-mode-improvements' into update/learning-mode-default-template

donnapep committed:
Fix linter issues after resolving conflicts

donnapep committed:
Fix issue with module name after resolving conflicts

donnapep committed:
Fix colors of lessons in sidebar

donnapep committed:
Update styles for different button states

donnapep committed:
Fix module name styles in Course theme

donnapep committed:
Eliminate $sensei-text SASS variable

donnapep committed:
Put CSS variables into :root selector

donnapep committed:
Style course title

donnapep committed:
Undo change to put CSS variables in :root selector

donnapep committed:
Remove :root selector as it prevents customizing styles in Course theme

donnapep committed:
Fix module title size on mobile for Course theme

donnapep commented:
@merkushin This is ready for another review. To test, pull down the latest changes from [this branch](https://github.com/Automattic/themes/pull/7008). Changes include:

- Button hover and focus states are per Andrei's designs.
- Styling the course title
- Fixing the module title size in the sidebar on the Course theme for mobile and desktop (should always be `28px`).

## Button Testing Instructions
- Add a _Contact Teacher_ block to a lesson.
- Add a Buttons block with one button using the Fill style and the another button using the Outline style.
- Check to ensure that the above buttons, as well as the _Complete Lesson_, _Take Quiz_, and _Completed_ buttons all use the hover and focus styles from the designs.

donnapep committed:
Fix media query

merkushin commented:
Tested on desktop and mobile.
Typography and colors look good :) 

I found two issues.

1. With "Next Lesson >" and "View Quiz >" links. These links don't follow the proposed styles. Should they?
<img width="478" alt="CleanShot 2023-04-13 at 09 01 53@2x" src="https://user-images.githubusercontent.com/329356/231621033-e30e713e-720c-4b9b-916a-be1f2652f466.png">
<img width="280" alt="CleanShot 2023-04-13 at 09 04 34@2x" src="https://user-images.githubusercontent.com/329356/231621058-ee18b8a1-ecf2-4751-ae6f-d2297b201f52.png">

2. Pagination. For pagination, we need a separate card (#6806) for putting it in the right place and remove duplicates. But here I also want to show it doesn't follow our color guidelines.
<img width="478" alt="CleanShot 2023-04-13 at 08 54 43@2x" src="https://user-images.githubusercontent.com/329356/231621115-60b4c881-c913-4660-a8c5-919506e3fb2a.png">

I wanted to notice it might worth addressing these issues in another PR.

merkushin commented on assets/css/sensei-course-theme/base.scss:
```diff
@@ -1,27 +1,22 @@
 	--wp--style--block-gap: 0px;
 	box-sizing: border-box;
+	color: $sensei-module-lesson;
+	font-size: 13px;
```
Should we use `0.8125rem` instead?

donnapep replied:
Fixed in https://github.com/Automattic/sensei/pull/6783/commits/e9ca39ea7b403486b9751a601d1b706210a53d79

gabrielcaires commented on assets/css/sensei-course-theme/blocks/course-title.scss:
```diff
@@ -1,9 +1,9 @@
-:root .wp-block-sensei-lms-course-title {
+.wp-block-sensei-lms-course-title {
+	color: #000000;
```
```suggestion
	color: #1E1E1E;
```
According the designs

donnapep replied:
It's black as per the designs I'm following here - https://www.figma.com/file/8mY8ui4zYK1Ylw2yIFi9Lx/Learning-Mode?node-id=20020527-41973&t=wJvNPrH0b40P4Ljq-0

gabrielcaires replied:
I double-checked with Andrei, we should use gray p1681396372830909-slack-C02NWDZBL0H

donnapep replied:
Fixed in https://github.com/Automattic/sensei/pull/6783/commits/a829ff887961a4dc85e634966fe827a91c92bd94.

gabrielcaires commented on assets/css/sensei-theme-blocks.scss:
```diff
@@ -7,5 +7,6 @@
 @import 'sensei-course-theme/blocks/course-title';
 @import 'sensei-course-theme/blocks/exit-course-button';
 @import 'sensei-course-theme/blocks/contact-teacher';
+@import 'sensei-course-theme/blocks/exit-course';
```
We already have another file to customize it, https://github.com/Automattic/sensei/blob/f12e3f9dde65609be1ee1f51039b14fbc6692a78/assets/css/sensei-course-theme/blocks/exit-course-button.scss

That is included  [here](https://github.com/Automattic/sensei/blob/f12e3f9dde65609be1ee1f51039b14fbc6692a78/assets/css/sensei-theme-blocks.scss#L8) 

Should we merge the selectors?

donnapep replied:
Fixed in https://github.com/Automattic/sensei/pull/6783/commits/da6e867ce3f25e5d0482d599ee3ffc3f15f5af6c/. Kept the file name that matched the block's name. The CSS was essentially equivalent.

donnapep replied:
Fixed in https://github.com/Automattic/sensei/pull/6783/commits/a829ff887961a4dc85e634966fe827a91c92bd94

gabrielcaires commented on assets/css/sensei-course-theme/colors.scss:
```diff
@@ -0,0 +1,12 @@
+$sensei-module-lesson: #101517;
```
I think we should use CSS variables here and centralize all of them on the lines below. It removes the responsibility to include this file.

donnapep replied:
Fixed in https://github.com/Automattic/sensei/pull/6783/commits/d481901ac06c00a7bdb031dc23030306fac2879b.

gabrielcaires commented on includes/blocks/course-theme/class-lesson-actions.php:
```diff
@@ -174,20 +174,18 @@ public function render( array $attributes = [] ): string {
 			// Complete button.
 			if ( ! $is_pass_required ) {
-				$complete_lesson_button = $this->render_complete_lesson( $complete_button_class, $has_incomplete_prerequisite );
+				$complete_lesson_button = $this->render_complete_lesson( 'is-secondary', $has_incomplete_prerequisite );
```
This method is private and is only used here, so I think we can remove this param.

gabrielcaires commented on includes/blocks/course-theme/class-lesson-module.php:
```diff
@@ -49,22 +50,36 @@ public function __construct() {
 
-		$wrapper_attr = get_block_wrapper_attributes();
+		// Determine the output class.
+		$class = 'sensei-course-theme-lesson-module';
```
Can we centralize all styles from this block using `.wp-block-sensei-lms-course-theme-lesson-module` ? so we can remove this custom logic.

donnapep replied:
Sure - https://github.com/Automattic/sensei/pull/6783/commits/60d7e44373c8c6f9eef25ae45468eec564b27a10

donnapep commented:
> With "Next Lesson >" and "View Quiz >" links. These links don't follow the proposed styles. Should they?

~I couldn't get the _Next Lesson_ link to display, so I thought perhaps we were hiding it or else there may be a bug somewhere. Is it always appearing for you? Which theme did you test on?~ Never mind. I figured out what I was doing wrong. 

I'll look into the _View Quiz_ link and pagination links for sure.

donnapep committed:
Style link buttons

gabrielcaires reviewed (changes requested):
I left some optional comments but I think the main we need to address is the editor is not showing the correct font sizes for the course title and module title.

Before the PR changes
<img width="1548" alt="Screenshot 2023-04-13 at 18 43 24" src="https://user-images.githubusercontent.com/38718/231841352-a342af49-d57b-4156-a2a5-8802cb789417.png">



After PR changes:
<img width="1583" alt="Screenshot 2023-04-13 at 18 42 14" src="https://user-images.githubusercontent.com/38718/231841076-4cbb9521-f593-46d6-9616-9d5132836e71.png">

I also noticed it is ignoring the colors from the global styles.

donnapep committed:
Switch to rem units

donnapep committed:
Update color of course title

donnapep committed:
Remove extraneous .scss file

The styles can be found in exit-course.scss instead

donnapep committed:
Hide theme's pagination

donnapep committed:
Make all colors CSS variables

donnapep commented:
> I left some optional comments but I think the main we need to address is the editor is not showing the correct font sizes for the course title and module title.

Because this card was never intended to address the editor, I'd rather leave editor issues for another card.

donnapep committed:
Still need to import colors in learning-mode-compat

donnapep committed:
Update pagination

donnapep committed:
Import colors in base CSS so that color vars can be used

donnapep committed:
Move CSS color vars back to base.scss since there's breakage potential

donnapep committed:
Fix global styles for course title

donnapep committed:
Fix global styles for lesson module

donnapep committed:
Update color of course title for classic themes

donnapep committed:
Fix global styles for module title and "Quiz" link in course navigation

donnapep committed:
Fix global styles for lessons in course navigation

donnapep committed:
Remove use of extraneous CSS class

donnapep committed:
Change Quiz element from span to anchor

donnapep committed:
Fix pagination markup in editor to match frontend

donnapep committed:
Update module title markup to match frontend

donnapep committed:
Remove obsolete CSS property

donnapep committed:
Wrap editor markup in .editor-styles-wrapper class

donnapep committed:
Undo wrapping CSS in .editor-styles-wrapper

donnapep committed:
Fix course title in editor

donnapep committed:
Editor - Progress counter color

donnapep committed:
Editor - Exit Course link

Partial fix; color not correctly applied in editor for classic themes due to separate issue

donnapep committed:
Fix global styles for pagination and module summary

donnapep committed:
Tweaks to buttons

donnapep commented:
@Automattic/nexus This PR is ready for another review. Here's what's changed since the last review:

- Pagination has been styled. To test this, add a page break block to a lesson.
- Global colors have been fixed. To test this, fetch the latest changes from [this branch](https://github.com/Automattic/themes/pull/7008), switch to the Course theme, and update all the global colors (links, headings etc.). Ensure lessons in LM use the appropriate global colors.
- Although I made a few updates for the editor, the editor fixes will be handled by a separate card.

There is one discrepancy between the designs and the UI for the Course theme. The lesson and Quiz links in the sidebar have not been updated to new colors. After some discussion p1681843345827139-slack-C013QUH20TS, we're keeping them at `#00594F`.

Be sure to run `npm run start` when testing.

For small issues, I'd prefer to open separate issues in order to get this PR merged as I think these changes will be needed by everyone. So please don't hold back approval for small things, only major problems. 🤞🏻

gabrielcaires commented on assets/course-theme/blocks/lesson-blocks/index.js:
```diff
@@ -212,14 +212,12 @@ export default [
 		edit: function EditPageActions() {
 			const blockProps = useBlockProps( {
-				className: 'sensei-quiz-pagination',
+				className: 'sensei-course-theme__post-pagination',
```
Maybe we can think about removing this customs classes and using the generated by the block

donnapep replied:
Yeah, I agree. I don't understand the reason for adding additional custom classes. Although I'm not fixing it here, I'll keep it in mind for future PRs.

gabrielcaires commented on assets/course-theme/blocks/lesson-blocks/module-title/module-title-edit.js:
```diff
@@ -7,5 +7,5 @@ import { useBlockProps } from '@wordpress/block-editor';
 	const blockProps = useBlockProps();
 
-	return <h3 { ...blockProps }>{ __( 'MODULE', 'sensei-lms' ) }</h3>;
+	return <div { ...blockProps }>{ __( 'Module', 'sensei-lms' ) }</div>;
```
Why are using div instead of h3?

donnapep replied:
I don't recall. Switched back [here](https://github.com/Automattic/sensei/pull/6783/commits/cfb327197d734f03839b497579a80af6180d4fed).

gabrielcaires reviewed (dismissed):
Working as described.
I just noticed the navigation bar is not using the correct font sizes, but I saw we are going to address it in a different PR.

Aprovado!  🇧🇷

donnapep committed:
Use h3 for lesson module

donnapep commented:
> I just noticed the navigation bar is not using the correct font sizes, but I saw we are going to address it in a different PR.

@gabrielcaires The sidebar should be using the correct font sizes already. Which theme and which element in particular? I'll check it out.
